qa/pull-tester/rpc-tests.sh

```diff
@@ -34,6 +34,7 @@ testScripts=(
     'maxblocksinflight.py'
     'invalidblockrequest.py'
     'rawtransactions.py'
+    'retarget-overflow.py'
```

Comment:
The rule for regtest is just to start with a much lower difficulty than
mainnet, but otherwise is identical in retargetting.

It's unfortunate that we would need larger integer support for just the
regtest.

As regtest mining beyond the retargetting interval hasn't worked for a
while, could we not just change its logic instead, and disable retargetting
entirely? That seems closer to what people expect and needs fewer code
changes to non-test code.

And yes, we do need a test that exercises retargetting... @sdaftuar?